File: projects/mobile-app-practice/01-notepad/app/src/main/java/cn/edu/android/notepad/MainActivity.java
package cn.edu.android.notepad;

import android.app.AlertDialog;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    private RecyclerView rvList;
    private ImageView ivAdd;
    private TextView tvMainTitle;
    private TextView tvEmpty;
    private NoteAdapter adapter;
    private List<NoteBean> list = new ArrayList<>();
    private DBUtils dbUtils;
    private ActivityResultLauncher<Intent> launcher;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initLauncher();
        init();
    }

    private void initLauncher() {
        launcher = registerForActivityResult(
                new ActivityResultContracts.StartActivityForResult(),
                result -> {
                    if (result.getResultCode() == 2) {
                        showQueryData();
                    }
                }
        );
    }

    private void init() {
        dbUtils = DBUtils.getInstance(this);
        tvMainTitle = findViewById(R.id.tv_main_title);
        tvEmpty = findViewById(R.id.tv_empty);
        rvList = findViewById(R.id.rv_list);
        ivAdd = findViewById(R.id.iv_add);
        tvMainTitle.setText(R.string.notepad_title);
        rvList.setLayoutManager(new LinearLayoutManager(this));

        adapter = new NoteAdapter(this, new NoteAdapter.ItemClickListener() {
            @Override
            public void onItemClick(View v, int position) {
                if (position < 0 || position >= list.size()) {
                    return;
                }
                NoteBean bean = list.get(position);
                Intent intent = new Intent(MainActivity.this, RecordActivity.class);
                intent.putExtra("id", bean.getId());
                intent.putExtra("title", bean.getTitle());
                intent.putExtra("content", bean.getContent());
                intent.putExtra("time", bean.getTime());
                launcher.launch(intent);
            }

            @Override
            public void onItemLongClick(View v, int position) {
                if (position < 0 || position >= list.size()) {
                    return;
                }
                showDeleteDialog(position);
            }
        });

        ivAdd.setOnClickListener(v -> launcher.launch(new Intent(MainActivity.this, RecordActivity.class)));
        rvList.setAdapter(adapter);
        showQueryData();
    }

    private void showDeleteDialog(int position) {
        new AlertDialog.Builder(MainActivity.this)
                .setIcon(R.drawable.ic_delete)
                .setTitle("删除提示")
                .setMessage("是否删除此记录？")
                .setPositiveButton("确定", (dialog, which) -> {
                    NoteBean noteBean = list.get(position);
                    if (dbUtils.deleteNote(noteBean.getId())) {
                        list.remove(position);
                        adapter.setData(list);
                        updateEmptyState();
                        Toast.makeText(MainActivity.this, "删除成功", Toast.LENGTH_SHORT).show();
                    } else {
                        Toast.makeText(MainActivity.this, "删除失败", Toast.LENGTH_SHORT).show();
                    }
                })
                .setNegativeButton("取消", (dialog, which) -> dialog.dismiss())
                .show();
    }

    private void showQueryData() {
        list = dbUtils.queryNote();
        adapter.setData(list);
        updateEmptyState();
    }

    private void updateEmptyState() {
        tvEmpty.setVisibility(list == null || list.isEmpty() ? View.VISIBLE : View.GONE);
    }
}

File: projects/03-mall-list/app/src/main/java/com/example/malllist/MainActivity.kt
package com.example.malllist

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    private val goodsList = listOf(
        Goods(
            "针织毛衣",
            "加厚保暖针织毛衣，面料柔软亲肤，适合秋冬日常通勤穿搭。",
            R.drawable.wireclothes
        ),
        Goods(
            "户外圆桌",
            "复古风格小圆桌，阳台与庭院都适用，适合下午茶和休闲会客。",
            R.drawable.table
        ),
        Goods(
            "轻盈围巾",
            "渐变色系长款围巾，垂坠感自然，春秋季可作穿搭点缀。",
            R.drawable.scarf
        ),
        Goods(
            "奇异果礼盒",
            "新鲜奇异果酸甜多汁，富含维生素，适合作为家庭水果补给。",
            R.drawable.kiwifruit
        ),
        Goods(
            "奶油蛋糕",
            "奶油裱花蛋糕造型精致，适合生日聚会、纪念日庆祝等场景。",
            R.drawable.cake
        ),
        Goods(
            "红苹果",
            "果形饱满、口感清甜爽脆，适合作为日常健康零食和早餐搭配。",
            R.drawable.apple
        )
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val recyclerView: RecyclerView = findViewById(R.id.recycler_view)
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = GoodsAdapter(goodsList)
    }
}
